src/rust/ensogl/lib/components/src/scrollbar.rs

```diff
@@ -0,0 +1,152 @@
+            track_position_upper.target <+ frp.set_track.map(|b| b.end);
+            track_position_lower.skip   <+ frp.set_track.constant(());
+            track_position_upper.skip   <+ frp.set_track.constant(());
+
+            // Normalise values for internal use.
+            normalised_track_bounds <- all2(&frp.track,&frp.set_overall_bounds).map(|(track,overall)|{
+                 Bounds::new(normalise_value(&(track.start,*overall)),normalise_value(&(track.end,*overall)))
+            });
+            normalised_track_center <- normalised_track_bounds.map(|bounds| bounds.center());
+
+            // Slider Updates
+            update_slider <- all(&normalised_track_bounds,&frp.resize);
+            eval update_slider(((value,size)) model.set_background_range(*value,*size));
+
+            // Mouse IO - Clicking
+            click_delta <- base_frp.background_click.map2(&normalised_track_center,
+                f!([model](click_value,current_value) {
+                    let rotation  =  Rotation2::new(-model.rotation().z);
+                    let rotated   = rotation * click_value;
+                    let direction = if rotated.x > *current_value { 1.0 } else { -1.0 };
+                    const SCALE: f32 = 0.05;
+                    direction * SCALE
+            }));
+            click_animation <- click_delta.constant(true);
+
+            // Mouse IO - Dragging
+            drag_movement <- mouse.translation.gate(&base_frp.is_dragging_any);
+            drag_delta    <- drag_movement.map2(&base_frp.track_max_width, |delta,width|
+                (delta.x + delta.y) / width
+            );
+            drag_delta     <- drag_delta.gate(&base_frp.is_dragging_track);
+            drag_animation <- drag_delta.constant(false);
+
+            // Mouse IO - Event Evaluation
+            should_animate <- any(&click_animation,&drag_animation);
+
+            drag_center_delta <- any(&drag_delta,&click_delta);
+            drag_update       <- drag_center_delta.map2(&normalised_track_bounds,|delta,Bounds{start,end}|
+                Bounds::new(start+delta,end+delta)
+            );
+
+            is_in_bounds <- drag_update.map(|value| should_clamp_with_overflow(value,&None));
+
+            new_value_absolute <- all(&frp.set_overall_bounds,&drag_update).map(|(bounds,Bounds{start,end})|
+                Bounds::new(
+                    absolute_value(&(*bounds,*start)),absolute_value(&(*bounds,*end))).sorted()
+            ).gate(&is_in_bounds);
+
+            new_value_lower <- new_value_absolute.map(|b| b.start);
+            new_value_upper <- new_value_absolute.map(|b| b.end);
+
+            track_position_lower.target <+ new_value_lower.gate(&is_in_bounds);
+            track_position_upper.target <+ new_value_upper.gate(&is_in_bounds);
+
+            track_position_lower.skip <+ should_animate.on_false();
+            track_position_upper.skip <+ should_animate.on_false();
+
```

Comment:
This looks like pretty complex FRP - it seems it includes dragging etc. Why such things as dragging are not completely handled by slider instead? I was thinking that this component would just wrap slider and just set a few FRP values to it, without its own complex FRP utility. Also, there are no docs describing what really these functions do, which makes it much harder to understand - it contains some dragging, some mouse IO, while all of these things should be handled by the slider component.

Reply:
The actual drag implementation and underlying logic is shared. But this is the logic how to interpret it, which is different in each component: (1) single number selectors have clicking at the target and dragging only on the background (2) range slider need to act on dragging either the track piece or the edges of the track piece (3) the scrollbar needs to have dragging of the track piece and "jumping" when clicking on some other part of the background. 
That is essentially what is encoded here. Maybe some things could be refactored and abstracted, but not too much.

Reply:
Why cant these things be just coded in (moved to) the slider implementation so you'd be able to choose which mode you wants with FRP settings? Just like choosing colors, we could tell "I want this slider to jump to the place when clicking the background". This should be part of the sldier configuration rather than extracted here imo. Can we move it there, please?